src/Controller/CrudController.php

```diff
@@ -301,31 +327,44 @@ public function exportAction(Request $request, TableFactory $tableFactory, Table
             }
         }
 
-        $this->getDefinition()->configureExport($table);
-        $this->getDefinition()->configureFilters($table);
-        $this->getDefinition()->configureTableExporter($table);
-        if ($request->query->getInt('all', 0) === 1) {
-            $table->getExtension(PaginationExtension::class)?->setLimit(0);
-        }
-        $exporter = $table->getExporter($request->query->getString('exporter', 'table'));
-        if (!$exporter && count($table->getExporters()) > 0) {
-            $exporter = $table->getExporter(key($table->getExporters()));
-        }
-        if (!$exporter instanceof ExporterInterface) {
-            $this->addFlash('error', 'araise_crud.export_error');
-            throw new \RuntimeException('No Exporter found.');
-        }
-        $spreadsheet = $exporter->createSpreadsheet($table);
-        $writer = new Xlsx($spreadsheet);
-        $response = new StreamedResponse();
-        $response->setCallback(
-            function () use ($writer) {
-                $writer->save('php://output');
+        assert($entity !== null);
```

Comment:
On a scale from 1-10, how evil is this?
Because if this doesn't exist the code above should throw a NotFoundHttpException, I think... So this shouldn't be an issue. But please take a look at the whole code @tuxes3 - just to make sure 😅

Reply:
It wouldn't be too evil, if it would make sense. The Entity is only needed in the `RelationContent` Export case which @ArnoEgli developed. In the standard case there is no need for an `entity`. Please compare the implementation on `develop` with yours. There is no need to have the `entity` if `acronym` or `entityId` is not set.

Reply:
These lines [just changed](https://github.com/araise-dev/CrudBundle/pull/66/files) anyways. So I'm gonna take the develop version of this function and try it again. Thanks for the feedback 👍